### PR TITLE
VEN-741, VEN-742, VEN-743, VEN-744 | Unmarked winter storage form ("Nostojärjestysilmoitus")

### PR DESCRIPTION
**Pull request by TatuArvela**
## Description

* Implement unmarked winter storage form
* Improve unmarked winter storage choose area page
* Refactor shared form components
  * to allow an arbitrary number of application steps before form
  * to allow using different legends for form pages between forms

## Related issues

* [VEN-741](https://helsinkisolutionoffice.atlassian.net/browse/VEN-741): Unmarked WS areas: implement choose area page
* [VEN-742](https://helsinkisolutionoffice.atlassian.net/browse/VEN-742): Unmarked WS areas: implement boat's information page
* [VEN-743](https://helsinkisolutionoffice.atlassian.net/browse/VEN-743): Unmarked WS areas: implement customer's information page
* [VEN-744](https://helsinkisolutionoffice.atlassian.net/browse/VEN-744): Unmarked WS areas: implement overview page

### CODE REVIEW

File: src/components/forms/sections/WinterBoatDetails.tsx [hunk 1]
```diff
@@ -10,15 +10,39 @@ import WinterUnRegisteredBoat from '../tabs/WinterUnRegisteredBoat';
 import { WinterStorageMethod } from '../../../__generated__/globalTypes';
 import { WinterFormValues } from '../../../types/winterStorage';
 import { WithBoatType } from '../Selects';
-import SectionSelector from './SectionSelector';
+import SectionSelector, { TypeProps } from './SectionSelector';
```
Comment (santtul) [inline src/components/forms/sections/WinterBoatDetails.tsx:13]: It seems that this paragraph is missing:
Screenshot 2020-08-31 at 14 22 12

File: src/components/forms/sections/UnmarkedWinterOverview.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { WithBoatType } from '../Selects';
+
+type Props = {
+  values?: UnmarkedWinterFormValues;
+  selectedArea: WinterStorageArea;
+  boatTab: string;
+  steps: StepType[];
+} & WithBoatType;
+
+const UnmarkedWinterOverview = ({ values, selectedArea, boatTab, boatTypes, steps }: Props) => {
```
Comment (santtul) [inline src/components/forms/sections/UnmarkedWinterOverview.tsx:20]: these headers deviate from Abstract, they are defined in Abstract as "Talvisäilytysalue" and "Omistajan tiedot":
Screenshot 2020-08-31 at 14 30 27

File: src/components/pages/unmarkedWinterStoragePage/UnmarkedWinterStoragePage.tsx [hunk 3]
```diff
@@ -110,15 +114,15 @@ const UnmarkedWinterStoragePage = ({
...
-      <Container>
-        <Row>
-          <Col lg={{ size: 10, offset: 1 }} xl={{ size: 8, offset: 2 }}>
-            <div className="vene-unmarked-winter-storage-page__buttons">
-              <Button className="vene-unmarked-winter-storage-page__button">Jatka</Button>
-            </div>
-          </Col>
-        </Row>
-      </Container>
+      <div className="vene-unmarked-winter-storage-page__buttons">
```
Comment (santtul) [inline src/components/pages/unmarkedWinterStoragePage/UnmarkedWinterStoragePage.tsx:117]: the button could take the whole width in mobile (like in the next step):
Screenshot 2020-08-31 at 14 39 31

File: src/components/forms/sections/UnmarkedWinterOverview.tsx (hunk 2 above)
Comment (TatuArvela) [reply src/components/forms/sections/UnmarkedWinterOverview.tsx:20]: Good catch, thanks! I had this in mind, but looks like I forgot it.

File: src/components/forms/sections/WinterBoatDetails.tsx [hunk 4]
```diff
@@ -10,15 +10,39 @@ import WinterUnRegisteredBoat from '../tabs/WinterUnRegisteredBoat';
...
 
 type Props = {
   values?: WinterFormValues;
   tab: string;
+  requireBoat?: boolean;
+  showStorageMethod?: boolean;
 } & WithBoatType;
 
-const BoatDetails = ({ values, tab, boatTypes }: Props) => {
+const BoatDetails = ({ values, tab, requireBoat, boatTypes, showStorageMethod = true }: Props) => {
```
Comment (santtul) [inline src/components/forms/sections/WinterBoatDetails.tsx:22]: Could we rename this to `WinterBoatDetails`?

File: src/components/legends/unmarkedWinterStorageLegend/UnmarkedWinterStorageLegend.tsx [hunk 5]
```diff
@@ -19,11 +23,7 @@ const UnmarkedWinterStorageLegend = ({ legend, form, steps }: UnmarkedWinterStor
       <Container>
         <Row>
           <Col lg={{ size: 10, offset: 1 }} xl={{ size: 8, offset: 2 }}>
-            {steps && (
-              <div className="vene-unmarked-winter-storage-legend__steps">
-                <Steps steps={steps} />
-              </div>
-            )}
+            {steps && <Steps steps={steps} />}
```
Comment (santtul) [inline src/components/legends/unmarkedWinterStorageLegend/UnmarkedWinterStorageLegend.tsx:26]: Do we need these defensive checks for `steps`, `legend` and `form` ? seems that they are always given?

**Review by santtul — CHANGES_REQUESTED** [review]
Good stuff! Some comments there.

File: src/components/forms/sections/WinterBoatDetails.tsx (hunk 1 above)
Comment (TatuArvela) [reply src/components/forms/sections/WinterBoatDetails.tsx:13]: Added, also to the list of texts to be translated

File: src/components/pages/unmarkedWinterStoragePage/UnmarkedWinterStoragePage.tsx (hunk 3 above)
Comment (TatuArvela) [reply src/components/pages/unmarkedWinterStoragePage/UnmarkedWinterStoragePage.tsx:117]: Done 👍📱

File: src/components/forms/sections/WinterBoatDetails.tsx (hunk 4 above)
Comment (TatuArvela) [reply src/components/forms/sections/WinterBoatDetails.tsx]: Yes! ✏️

File: src/components/legends/unmarkedWinterStorageLegend/UnmarkedWinterStorageLegend.tsx (hunk 5 above)
Comment (TatuArvela) [reply src/components/legends/unmarkedWinterStorageLegend/UnmarkedWinterStorageLegend.tsx]: Good point, removed

**Review by santtul — APPROVED** [review]
giphy

**Pull request merged**